src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = []; // Масив для зберігання всіх живих тварин
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this); // Додаємо нову тварину до масиву живих
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true; // Травоїдна тварина ховається
+  }
+
+  unhide() {
+    this.hidden = false; // Травоїдна тварина виходить із сховку
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Herbivore) || animal.hidden) {
```

Comment:
The condition here ensures that the `bite` method only affects instances of `Herbivore` that are not hidden. This logic is correct according to the task requirements.